### PR TITLE
feat: Feast AWS Athena offline store (again)

### PR DESCRIPTION
**Pull request by toping4445**
What this PR does / why we need it:

It enables Feast users to use **S3+AWS Athena** as an offline store.

1. All unit tests passed (except for tests in test_sql_registry.py.) 
The tests above failed in my environment because of MySQLdb package-M1 compatibility issues.
2. some integration tests passed. (passed:109, failed:19, ignored:16)
I didn't implement some methods related to feature write. and tests with fixed S3 bucket name and related to GCP failed.
However, all tests in test_universal_historical_retrieval.py and test_univeral_types.py related to feature extraction have passed.

Which issue(s) this PR fixes:

Fixes #

Does this PR introduce a user-facing change?:

AWS Users can choose between Redshift and S3+Athena for an offline store.

### CODE REVIEW

**Comment by adchia** [comment]
I noticed there were a bunch of weird merge artifacts so I fixed them. Also removed Athena from the integration tests that run as part of CI since this is a contrib plugin.

can you sign your commits? None of them seem signed. 

Also: I have no way of reproducing the test results because I don't have Athena setup. It's ok for this PR, but maybe leave a comment in the Makefile to specify that it needs the user to have their own custom athena setup?

File: sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+        )
+
+        if table is None and query is None:
+            raise ValueError('No "table" argument provided.')
+        _name = name
+        if not _name:
+            if table:
+                _name = table
+            else:
+                warnings.warn(
```
Comment (adchia) [inline sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py:72]: can you remove this? It's past Feast 0.21 now haha so should enforce there is a name

**Review by adchia — APPROVED** [review]

File: sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py (hunk 1 above)
Comment (toping4445) [reply sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py]: I changed AthenaSource to have a name!

**Comment by adchia** [comment]
it’d be good to have some documentation, but this is good for an initial cut. Thanks!

**Comment by toping4445** [comment]
Is there a documentation guide? If you share it, I will refer to it and fill it out when I have time.

**Pull request merged**